File: day2.cpp
#include <algorithm>
#include <cstdio>
#include <fstream>
#include <iostream>
#include <string>

int main() {
    std::ifstream file("day2.input");

    long long totalPaper = 0;
    long long totalRibbon = 0;

    std::string line;

    while (std::getline(file, line)) {
        if (line.empty()) continue;

        int l, w, h;

        std::sscanf(line.c_str(), "%dx%dx%d", &l, &w, &h);

        int area = 2 * (l * w + w * h + l * h);
        int extra = std::min({l * w, w * h, l * h});
        totalPaper += area + extra;

        int perimeter = 2 * (l + w + h - std::max({l, w, h}));
        int bow = l * w * h;

        totalRibbon += perimeter + bow;
    }

    std::cout << totalPaper << "\n";
    std::cout << totalRibbon << "\n";
}

File: day2.js
const fs = require("fs");

const lines = fs.readFileSync("day2.input", "utf-8").split("\n");

let totalPaper = 0;
let totalRibbon = 0;

for (const line of lines) {
    if (!line.trim()) continue;

    const [l, w, h] = line.split("x").map(Number);

    const area = 2 * (l * w + w * h + l * h);
    const extra = Math.min(l * w, w * h, l * h);
    totalPaper += area + extra;

    const perimeter = 2 * (l + w + h - Math.max(l, w, h));
    const bow = l * w * h;
    totalRibbon += perimeter + bow;
}

console.log(totalPaper);
console.log(totalRibbon);
